hips/hip-9999.md

```diff
@@ -0,0 +1,220 @@
+Centralizing the desired version guarantees that all users, automation, and libraries that use the package manager also honor the version locks in place.
+
+Making something native in helm that supports locking will be critical as users consume helm charts via other tools and workflows (e.g. gitops via `flux`).
+It will reduce errors because `helm` commands can be run from any users machine with access to the cluster.
+This is opposed to a helm plugin model where a user or tool may not have the plugin installed.
+
+Helm releases do not have the same constraints has traditional package managers.
+A cluster can have multiple versions of a chart deployed in different namespaces and each release can have various dependencies.
+Because releases are self-contained on a per-release basis, version locking information should also exist within the release metadata and be read and processed during template rendering.
+
+## Specification
+
+To implement version locking on a release an optional field should be added to the release secret that specifies a desired version.
+
+```
+apiVersion: v1                                                           
+data:                                                                 
+  release: SDRzSUF...
+kind: Secret
+metadata:
+  creationTimestamp: "2021-02-23T18:07:25Z"
+  desiredChartVersion: "8.0.*"             <---- this field is added
+  labels:
+    modifiedAt: "1614103661"
+    name: ingress
+    owner: helm
+    status: deployed
+    version: "1"
+  name: sh.helm.release.v1.ingress.v1
+  namespace: default
+  resourceVersion: "47331598"
+  selfLink: /api/v1/namespaces/default/secrets/sh.helm.release.v1.ingress.v1
+  uid: 07ba4723-1cd1-43a8-a322-7d8e25f0868e
+type: helm.sh/release.v1
+```
+_Field name and exact json path may change_
+
+Having a desired chart version will allow customers to stay on specific chart versions based on the published chart version data.
+The chart field is required to be SemVer 2 version and the `desiredChartVersion` can be an exact version match (eg `1.0.0`) or a wildcard string match (e.g. `1.*`).
+
+The field can be added with the `helm lock` command.
+It will take a release name and chart version to lock to.
+```
+helm lock autoscaler --version '8.*'
+```
+_The version is a free form field but should be validated aganist SemVer standards with `*` wildcards_
+
+Users can also manually modify the secret to add the field, but that is not advised.
+
+Locks can be seen in the `helm list` command.
+```
+NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                                  APP VERSION
+autoscaler      default         1               2021-02-23 17:58:07.479425206 +0000 UTC deployed        cluster-autoscaler-8.0.0 (locked 8.*)  1.17.1
+```
+Current chart versions can be seen from the `helm show chart [CHART]` command.
```

Comment:
Note that under current backward compatibility rules, this is a breaking change to add a new field to `helm list`'s output. Would need a flag in Helm 3, and could become default in Helm 4.

Reply:
I was hoping the change would be backwards compatible but I am not familiar enough with breaking changes with helm v2. If you think it requires waiting for helm 4 that is fine. I hope it can be considered and included at some point.